tensornetwork/network_operations.py

```diff
@@ -131,18 +132,17 @@ def copy(nodes: Iterable[BaseNode],
     node1 = edge.node1
     axis1 = edge.node1.get_axis_number(edge.axis1)
 
-    if not edge.is_dangling():
+    if not edge.is_dangling() and edge.node2 in node_dict:
```

Comment:
It might be the case that the edge is only connected to a node in the node set via `edge.node2`. I'm not sure that's handled here.

Reply:
no it wasn't handled. didn't notice, fixed it now, thanks!